src/wallet/spvscanner.cpp

```diff
@@ -131,24 +155,42 @@ void CSPVScanner::RequestBlocks()
     }
 
     // skip blocks that are before startTime
-    const CBlockIndex* skip = lastProcessed;
+    CBlockIndex* skip = lastProcessed;
     while (skip->GetBlockTime() < startTime && partialChain.Height() > skip->nHeight)
+    {
         skip = partialChain.Next(skip);
-    if (skip != lastProcessed) {
+    }
+    if (skip != lastProcessed)
+    {
         LogPrint(BCLog::WALLET, "Skipping %d old blocks for SPV scan, up to height %d\n", skip->nHeight - lastProcessed->nHeight, skip->nHeight);
         UpdateLastProcessed(skip);
-        if (lastProcessed->nHeight > requestTip->nHeight) {
+        if (lastProcessed->nHeight > requestTip->nHeight)
+        {
             requestTip = lastProcessed;
         }
     }
 
     std::vector<const CBlockIndex*> blocksToRequest;
 
-    // add requests for as long as nMaxPendingRequests is not reached and there are still heigher blocks in headerChain
-    while (requestTip->nHeight - lastProcessed->nHeight < MAX_PENDING_REQUESTS &&
-           partialChain.Height() > requestTip->nHeight) {
-
+    // add requests for as long as nMaxPendingRequests is not reached and there are still higher blocks in headerChain
+    // In the special case of 'skipped' blocks we allow a higher restriction - as they don't represent real network processing so won't starve peers
+    int nNumSkipped=0;
+    while ((requestTip->nHeight - lastProcessed->nHeight - nNumSkipped < MAX_PENDING_REQUESTS) &&
+          (requestTip->nHeight - lastProcessed->nHeight < 5000) &&
+          (partialChain.Height() > requestTip->nHeight)
+          )
+    {
         requestTip = partialChain.Next(requestTip);
+        if (CanSkipBlockFetch(requestTip, Checkpoints::LastCheckPointHeight()))
+        {
+            requestTip->nStatus |= BLOCK_VALID_MASK;
```

Comment:
Is the priorty queue the best way to do this (it seemed the least intrusive way in terms of code changes) - but maybe theres a cleaner way?